scrapper/src/main/java/edu/java/scrapper_body/clients/github_client/GitHubCommitUpdateClient.java

```diff
@@ -0,0 +1,96 @@
+package edu.java.scrapper_body.clients.github_client;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import edu.java.scrapper_body.clients.unsupported_client.UnsupportedClient;
+import edu.java.scrapper_body.clients_body.AbstractClient;
+import edu.java.scrapper_body.clients_body.Client;
+import edu.java.scrapper_body.clients_body.Response;
+import java.net.URI;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import org.springframework.http.MediaType;
+import org.springframework.web.reactive.function.client.WebClient;
+
+public final class GitHubCommitUpdateClient extends AbstractClient {
+    private static final String BASE_URI = "https://api.github.com";
+    private static final String HOST = "github.com";
+    private static final String COMMIT = "commit";
+    private static final String COMMITTER = "committer";
+
+    public GitHubCommitUpdateClient(String baseURI, Client nextClient) {
+        super(WebClient.create(baseURI), nextClient);
+    }
+
+    public GitHubCommitUpdateClient(Client nextClient) {
+        this(BASE_URI, nextClient);
+    }
+
+    public GitHubCommitUpdateClient(String baseURI) {
+        this(baseURI, new UnsupportedClient());
+    }
+
+    public GitHubCommitUpdateClient() {
+        this(BASE_URI, new UnsupportedClient());
+    }
+
+    @Override
+    public List<Response> newUpdates(URI uri) throws JsonProcessingException {
+        if (notValid(uri)) {
+            return nextClient.newUpdates(uri);
+        }
+
+        String json = webClient
+            .get()
+            .uri("/repos" + uri.getPath() + "/commits")
+            .accept(MediaType.APPLICATION_JSON)
+            .retrieve()
+            .bodyToMono(String.class)
```

Comment:
можно парсить сразу в List<Commit>. тогда страшный метод commitsFromJSON можно будет удалить :)

Reply:
касается обоих клиентов

Reply:
Не очень понял, как это сделать. Я хочу получать конкретные поля из json и формировать новый объект с ними, а автоматически в лист он так не парсит. С гитхабом проблема решилась с помощью одно десериализатора. А со стековерфлоу не очень получилось из-за формата приходящего json файла.

`{
    "items": [
        {
            "owner": {
                "account_id": 881430,
                "reputation": 360,
                "user_id": 263597,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/d4d90f857b84617ffff47212595dcfe9?s=256&d=identicon&r=PG",
                "display_name": "gbuys",
                "link": "https://stackoverflow.com/users/263597/gbuys"
            },
            "is_accepted": true,
            "score": 2,
            "last_activity_date": 1708607611,
            "creation_date": 1708607611,
            "answer_id": 78041218,
            "question_id": 78040876,
            "content_license": "CC BY-SA 4.0",
            "body": "<p>It seems like you should use jaxb-maven-plugin instead of maven-jaxb2-plugin. See <a href=\"https://github.com/highsource/jaxb-tools?tab=readme-ov-file#jaxb-maven-plugin\" rel=\"nofollow noreferrer\">github</a></p>\n"
        }
    ],
    "has_more": false,
    "quota_max": 300,
    "quota_remaining": 299
}`

Я хочу получать информацию из items, а в лист он не читает из-за последних трех полей. Как сделать лучше не нашел.

Reply:
`            .bodyToMono(StackOverflowResponse.class)
`
В классе StackOverflowResponse:
```
public class StackOverflowResponse {

    @JsonProperty("items")
    List<Item> items;
}
```

Item:
```
public class Item {

    OffsetDateTime dateTime;

    @JsonProperty("last_activity_date")
    private void parseDate(long seconds) {
        dateTime = OffsetDateTime.ofInstant(Instant.ofEpochSecond(seconds), ZoneOffset.UTC);
    }
}
```

Это лишь пример. В общем - используй @JsonProperty и будешь мапить что нужно

Reply:
Хорошо, попробую так сделать. Спасибо за наводку